.github/workflows/build-python-packages.yml

```diff
@@ -105,6 +105,11 @@ jobs:
       - name: Fully cleanup the toolcache directory before testing
         run: ./helpers/clean-toolcache.ps1 -ToolName "Python"
 
+      - name: Delete macOS /Library/Frameworks/Python.framework
+        if: matrix.platform == 'darwin'
+        shell: bash
+        run: if [ -d /Library/Frameworks/Python.framework ]; then sudo rm -rf /Library/Frameworks/Python.framework; fi
+
```

Comment:
This is required because current macos-11 runners come with 2 outdated pip dist-info.

When installing from pkg, these will be left alone and only one out of the 2 outdated version will be removed by `--upgrade` thus failing the tests.

Once runners are using fixed installs of Python, this block can removed.